containers/src/Data/IntMap/Internal.hs

```diff
@@ -3549,19 +3608,20 @@ mask i m
 
 -- | The prefix of key @i@ up to (but not including) the switching
 -- bit @m@.
-maskW :: Nat -> Nat -> Prefix
+maskW :: Nat -> Nat -> Int
 maskW i m
   = intFromNat (i .&. ((-m) `xor` m))
 {-# INLINE maskW #-}
 
--- | Does the left switching bit specify a shorter prefix?
-shorter :: Mask -> Mask -> Bool
-shorter m1 m2
-  = (natFromInt m1) > (natFromInt m2)
-{-# INLINE shorter #-}
+-- | Whether the left prefix is shorter than the prefix length indicated by the
+-- right mask.
+shorterMask :: Prefix -> Int -> Ordering
+shorterMask p1 m2 = natFromInt m2 `compare` natFromInt (px1 .&. (-px1))
+  where px1 = unPrefix p1
+{-# INLINE shorterMask #-}
 
 -- | The first switching bit where the two prefixes disagree.
-branchMask :: Prefix -> Prefix -> Mask
+branchMask :: Int -> Int -> Int
 branchMask p1 p2
   = intFromNat (highestBitMask (natFromInt p1 `xor` natFromInt p2))
```

Comment:
```suggestion
  = intFromNat (highestBitMask (natFromInt (p1 `xor` p2)))
```

Reply:
Actually, I think you can do better; change `highestBitMask` to work with `Int` instead of `Word`. `IntSet` will have to be adjusted, but you're planning to change that anyway.

Reply:
How do you mean "better"?
In terms of efficiency, it should not make a difference. Int <-> Word is free.
In terms of clarity, I agree that changing everything to work on Int would be good. I don't know why the Word/Nat stuff was done, maybe it made sense at the time.

Reply:
afaik `Word` is clearer when dealing with bits. `Int` is better for signed integers

Reply:
Converting between `Int` and `Word` is free at runtime. But those conversions have to be carried around in Core, and as far as I know they count towards unfolding "size" calculations used to guide inlining. So I don't think they're actually entirely free. Of course, that's probably also true to some extent for `newtype` casts, so maybe it's not worth worrying about?

Was some of the `Word` stuff done to catch type errors, or is it not disciplined enough for that?

Reply:
Anyway, simpler code is better, so streamlining things just to make it easier to read would be a plus, but not urgent.

Reply:
> afaik `Word` is clearer when dealing with bits. `Int` is better for signed integers

I disagree, there is no difference between `Word` and `Int` when treating them as a pattern of bits.

> ...as far as I know they count towards unfolding "size" calculations used to guide inlining

Interesting if that's true... all of these are marked INLINE though so it should be fine.

> Was some of the Word stuff done to catch type errors, or is it not disciplined enough for that?

I can't tell for sure, and there is no documented reason I can find. The Word stuff has been there since the beginning: https://github.com/haskell/containers/commit/bbbba97cbcf12039810533e3a2daf2eefdefe7f0

> Anyway, simpler code is better, so streamlining things just to make it easier to read would be a plus, but not urgent.

Agreed. If you're fine with it I can try removing the Word stuff after these changes are done to IntMap/IntSet.

Reply:
> I disagree, there is no difference between `Word` and `Int` when treating them as a pattern of bits.

When you designate some bytes as `Int` it means that specific bits serve to indicate whether the number is positive or negative usually by [Two's complement](https://en.wikipedia.org/wiki/Two%27s_complement). If you are just using a pattern of bits then the designation `Word` is correct.

Also you should verify that your logic works with a variable amount of bits because the size of `Int` is not guaranteed to be the same in all cases. If your logic depends on specific size then you should use Word8, Word16, Word32 or Word64.

Reply:
> When you designate some bytes as `Int` it means that specific bits serve to indicate whether the number is positive or negative usually by [Two's complement](https://en.wikipedia.org/wiki/Two%27s_complement).

This is true, but if we do not care about the _numeric value_, it doesn't matter. It is just some bits.

> Also you should verify that your logic works with a variable amount of bits because the size of `Word`/`Int` is not guaranteed to be the same in all cases. If your logic depends on specific size then you should use Word8, Word16, Word32 or Word64.

`IntMap` works with whatever the size of `Int` is.

Anyway, this is getting a little off-topic for this PR.